pylmod/base.py

```diff
@@ -0,0 +1,156 @@
+"""
+
+
+Defines the class StellarGradeBook
+"""
+
+import json
+import logging
+import requests
+
+
+log = logging.getLogger(__name__)  # pylint: disable=C0103
+
+
+class Base(object):
+    """
+    Python class representing interface to Stellar gradebook.
+
+    Example usage:
+
+    sg = StellarGradeBook('ichuang-cert.pem')
+    ats = sg.get('academicterms')
+    tc = ats['data'][0]['termCode']
+    sg.get('academicterm',termCode=tc)
+
+    students = sg.get_students()
+    assignments = sg.get_assignments()
+    sg.create_assignment('midterm1', 'mid1', 1.0, 100.0, '11-04-2013')
+
+    sid, student = sg.get_student_by_email(email)
+    aid, assignment = sg.get_assignment_by_name('midterm1')
+    sg.set_grade(aid, sid, 95.2)
+
+    sg.spreadsheet2gradebook(datafn)
+
+    """
+
+    GETS = {'academicterms': '',
+            'academicterm': '/{termCode}',
+            'gradebook': '?uuid={uuid}'}
+
+    GBUUID = 'STELLAR:/project/mitxdemosite'
+    TIMEOUT = 200  # connection timeout, seconds
+
+    verbose = True
+    gradebookid = None
+
+    def __init__(
+            self,
+            cert,
+            urlbase='https://learning-modules.mit.edu:8443/service/gradebook',
```

Comment:
I actually had that code in one my previous commits but took it out during one of the churns.  I'll add the issue and replace the code.